ext/geos_c_impl/geometry.c

```diff
@@ -1158,6 +1158,43 @@ method_geometry_polygonize(VALUE self)
   return result;
 }
 
+VALUE
+primary_method_geometry_voronoi_diagram(VALUE _,
+                                        VALUE self,
+                                        VALUE env,
+                                        VALUE tolerance,
+                                        VALUE only_edges)
+{
+  VALUE result;
+  const RGeo_GeometryData* self_data;
+  const GEOSGeometry* self_geom;
+  GEOSGeometry* diagram;
+  const GEOSGeometry* env_geom = NULL;
+
+  Check_Type(tolerance, T_FLOAT);
+
+  result = Qnil;
+  self_data = RGEO_GEOMETRY_DATA_PTR(self);
+  self_geom = self_data->geom;
+
+  if (self_geom) {
+    if (RB_TEST(env)) {
+      Check_TypedStruct(env, &rgeo_geometry_type);
+      env_geom = rgeo_convert_to_geos_geometry(self_data->factory, env, Qnil);
```

Comment:
This method could return NULL. In the spirit of v3, shouldn't we raise in stead?

Reply:
It is not, but that's my point! I think this debate should be part of another PR anyway. But here it can affect the behaviour:

I run `geom.voronoi_diagram(envelope: poly)`, and I expect the envelope to be `poly`, but since the conversion silently failed, the envelope param is actually ignored.

I can solve the local issue by raising in case of null value, and solve the global problem in a later PR. Or just do nothing here and accept the potential silent failure here, which is anyway not that likely to happen (`Feature.cast` would need to return nil)

Reply:
> I can solve the local issue by raising in case of null value, and solve the global problem in a later PR

I think this is the best approach. I agree that this behavior should be the standard since it's more obvious that there was an issue. Of course it's still a similar amount of work for the user since they have to rescue now instead of checking for `nil` but I think being more explicit and actually raising instead of silently failing is better.

Reply:
I have added a local security for now, I'm thinking of quickly remove it in favor of the global solution. Otherwise this error could be present in various places in the code base